Fix Migratable initializers so that they can never be run more than once

frangio opened the pull request:
Fixes #158.

facuspagnuolo commented on contracts/migrations/Migratable.sol:
```diff
@@ -29,17 +29,21 @@ contract Migratable {
     _;
     emit Migrated(contractName, migrationId);
     migrated[contractName][migrationId] = true;
+    migrated[contractName][INITIALIZED_ID] = true;
```
should we provide a way to store this information for on-chain contracts that have already been initialized? I can be done using a new empty initializer tho

frangio replied:
Oh, thank you. That was an important part of the fix that I forgot...

spalladino commented on contracts/migrations/Migratable.sol:
```diff
@@ -29,17 +29,21 @@ contract Migratable {
    */
   mapping (string => mapping (string => bool)) internal migrated;
 
+  string constant private INITIALIZED_ID = "initialized";
```
We're 100% sure that private constants are *not* assigned a slot in the storage, right? I've just checked with Remix and it does not seem to be the case.

frangio replied:
Experimentally, yes. But I don't think I've seen it documented...

We should probably look into it and push the Solidity team to document it because:

https://github.com/zeppelinos/zos-lib/blob/47c82a93d55fefa2f9067c2ea9313316aeda9083/contracts/upgradeability/UpgradeabilityProxy.sol#L24

frangio commented:
@facuspagnuolo @spalladino I added an initializer to `Migratable`, which was missing to take care of contracts that have already been deployed. I have to fix the tests but I need to do that a bit later today, will let you know.

frangio commented:
@spalladino @facuspagnuolo Please see the new changes. :slightly_smiling_face: 

Let me know if you think more tests are needed. I'm only testing that it's not possible to run two different initializers on the same contract. This means I'm not testing the original problem that we had seen, which was that an initializer could be run after an upgrade if it used a new migration id. I figured that the test I added covers the same situation and is simpler because it doesn't use a proxy.

In fact, I was a bit surprised to find that we're testing `Migratable` through a proxy, and not by itself. I don't think that's right for unit tests. If we're moving away from it, it's not worth the effort to change them, so I added a new section with the unit tests appropriate for this fix.
